src/main/core/sim_config.rs

```diff
@@ -449,5 +451,33 @@ fn verify_plugin_path(path: impl AsRef<std::path::Path>) -> anyhow::Result<()> {
         return Err(anyhow::anyhow!("The path is not executable"));
     }
 
+    // a cache so we don't check the same path multiple times (assuming the user doesn't move any
+    // binaries while shadow is running)
+    static CHECKED_DYNAMIC_BINS: Lazy<RwLock<HashSet<PathBuf>>> =
+        Lazy::new(|| RwLock::new(HashSet::new()));
+
+    let is_known_dynamic = CHECKED_DYNAMIC_BINS.read().unwrap().contains(path);
+
+    // check if the binary is dynamically linked
+    if !is_known_dynamic {
+        let ld_path = "/lib64/ld-linux-x86-64.so.2";
+        let ld_status = std::process::Command::new(ld_path)
+            .arg("--verify")
+            .arg(path)
+            .status()
+            .with_context(|| format!("Unable to run '{ld_path}'"))?;
+
+        if ld_status.success() {
+            CHECKED_DYNAMIC_BINS
+                .write()
+                .unwrap()
+                .insert(path.to_path_buf());
+        } else {
+            // technically ld-linux could return errors for other reasons, but this is the most
+            // likely reason given that we already checked that the file exists
+            return Err(anyhow::anyhow!("The path is not dynamically linked"));
```

Comment:
It might be helpful for debugging to include `ld.so`'s output. It looks like you can use `Command::output` instead of `Command::status` to also capture the output.

Reply:
As far as I know the `--verify` option doesn't output anything. For example if I give it a wrong path or the path to a static binary, there's nothing printed on stdout or stderr. But I'll add this since it doesn't hurt if they do add output in the future.